tinygrad/features/search.py

```diff
@@ -45,9 +45,10 @@ def _time_program(variables:List[Variable], outcount:int, rdev:Compiled, lib:byt
     if early_stop is not None and early_stop < tms[-1]: break
   return tms
 
-def _compile_linearizer(compiler:Compiler, lin:Linearizer, name:Optional[str]=None) -> Tuple[bytes, Optional[List[int]], Optional[List[int]],
-                                                                                             List[Variable], int, float, int]:
+def _compile_linearizer(compiler:Compiler, lin:Linearizer, name:Optional[str]=None, enforce_max:bool=False) \
+  -> Tuple[bytes, Optional[List[int]], Optional[List[int]], List[Variable], int, float, int]:
   lin.linearize()
+  if (uops_max:=getenv("BEAM_UOPS_MAX", 3000)) > 0 and len(lin.uops.uops) >= uops_max and enforce_max: raise Exception("too many uops")
```

Comment:
pick a more specific error than plain `Exception`

also put the predicate first, i would write something like
`if enforce_max and len(lin.uops.uops) >= getenv("BEAM_UOPS_MAX", 3000) > 0: raise ...`

Reply:
and this